core/test/test-helpers.js

```diff
@@ -85,6 +85,20 @@ function waitForChildListMutation(el) {
   });
 }
 
+function isElementVisible(el) {
+  if (!el) {
+    return false;
+  }
+  const styles = window.getComputedStyle(el);
+  return (
+    el.clientWidth !== 0 &&
```

Comment:
We should use `offsetWidth` instead of `clientWidth` (same for height), since it takes into account any scrollbars and border as well. And with this said, don't thing we need to check display for 'none', since if it is the case, then `offsetWidth` and `offsetHeight` will be 0.